### ISSUE TITLE
Double check your TeX

### ISSUE THREAD

**Issue by szymczdm** [JacquesCarette/literate-scientific-software#304]
After e639e56a2b46ec64d0b304a1967e5bca6125cdeb some of the TeX rendering has been modified. Please double check the typeset versions (pdfs) to ensure everything still looks alright. Namely: square brackets and parentheses, negations (boolean/integer), and inequalities.

Also, please ensure that you are typesetting the generated documents as you test, as I found some were not able to be typeset because of a minor bug in the TeX renderer.

As always, check it off when you're done and please close the issue once everyone is done. Thanks!

- [x] glassbr
- [x] gamephys
- [x] nopcm
- [x] swhs
- [x] ssp

**Comment by njericha** [comment]
I've tried to render my TeX file and it keeps crashing.

**Comment by szymczdm** [comment]
@njericha what exactly is going on? I'm rebuilding your example now and I'll test it as well.

**Comment by njericha** [comment]
It just says 
>TeXworks editor & previewer has stopped working
>A problem caused the program to stop working correctly. Windows will close the program and notify you if a solution is available.

**Comment by szymczdm** [comment]
That's odd. Seems like an issue with your editor. Just to be safe, try deleting all of the TeX generated files from the directory (ie. keep your .tex file, delete the pdf and log, aux, etc. files) then try to open it again. If it's still crashing, then there's most likely something wrong with your TeXworks installation.

**Comment by njericha** [comment]
Deleted the pdf and aux files and now it rendered!

**Comment by szymczdm** [comment]
@njericha  I just tried to render SSP and there's issues with missing "$" throughout. Namely because you have things like "stfMtrxDerivation" which doesn't use symbols, but instead is just hard-coded information (so your symbols like K*_i are not being rendered properly).

**Comment by njericha** [comment]
Not really an issue but spotted a difference in the "interstice normal/shear relationship" and "resistive shear" theoretical models. One has multiplication done by showing the variables beside, and the other one uses a "*".
![capture](https://user-images.githubusercontent.com/26230851/27483658-6bfb3060-57f4-11e7-814e-5b1a9f729281.PNG)
![capture2](https://user-images.githubusercontent.com/26230851/27483662-6fd3afb4-57f4-11e7-9f91-9c3fbac19cb6.PNG)

**Comment by AKM11** [comment]
This is the SRS directory TeX files you are referring to right? I have had issues with referencing for a while and cannot seem to generate a PDF.

**Comment by JacquesCarette** [comment]
@njericha Sometimes humans like to insert `*` for readability. In a setting like ours, this is really difficult to deal with.  For now, make multiplication implicit everywhere. We can try to allow this kind of tweaking later.  low priority.

**Comment by njericha** [comment]
In the original case study, there are no cases of `*` used for multiplication. This `*` shows up in Drasil generated files for some reason.

**Comment by JacquesCarette** [comment]
That's weird.  Sounds like something for @szymczdm to investigate.

**Comment by szymczdm** [comment]
@AKM11 you are likely missing the image files necessary for your example to typeset it. There's an open issue regarding automatically grabbing said files and moving them to the right location. You'll likely need to just grab the files from the SWHS case study (note: if there are any images that are in "pdf" format, you'll need to create a "png" version, since our HTML doesn't support embedding pdfs as images, and the TeX uses the same files as the HTML).

If you have trouble, let me know.

**Comment by szymczdm** [comment]
@njericha regarding the "*" difference, can you throw that in a different issue and I'll take a look shortly.

**Comment by AKM11** [comment]
@szymczdm Sorry, should have been more specific. Already have the PNG files, the warnings had something to do with referencing tables, sections, etc..  However, @hmadej  pointed out that this is more likely an invalid character issue that is preventing the program from running.  Working on it now.

**Comment by szymczdm** [comment]
@AKM11 Once you figure out what's going on, please let me know if it's an issue with the renderer or if it was just an issue in one of your example files.

**Comment by niazim3** [comment]
GlassBR has the following errors pop up when the Drasil generated code is run:
![image](https://user-images.githubusercontent.com/28247301/27484681-1efbaaf2-57f8-11e7-848b-b4dbfd468063.png)
Since stable is currently working fine, replacing the code with the following allows for a PDF to be generated: 
![image](https://user-images.githubusercontent.com/28247301/27484702-3634fa98-57f8-11e7-8a1e-422943bca01e.png)
I'm unsure of what I should be doing from here...

**Comment by szymczdm** [comment]
@niazim3 have you pulled the most recent changes and re-generated the .tex files? I can see from the error messages that it isn't using the changes I made (so if you're up to date, then this is a problem I'll need to hunt down).

**Comment by njericha** [comment]
The "%" comments out code in TeX

**Comment by niazim3** [comment]
@njericha Yes, those lines are commented out so that the errors are visible.
@szymczdm Yes, I have pulled the most recent version and regenerated the .tex file.

**Comment by szymczdm** [comment]
@niazim3 alright, I'll take a deeper look into it.

**Comment by AKM11** [comment]
@szymczdm Okay, at the moment, it seems that the issue is a result of using double subscripts on the same variable, e.g.:
![image](https://user-images.githubusercontent.com/28261578/27484682-20129220-57f8-11e7-9da0-40826a88fa7a.png)
(Note the reference warnings I was looking at the first time)
In the Case Study, variables with a subscript in addition to a max/min denotation have the word max/min written as superscript, while variables without subscripts have max/min written as subscripts.
![image](https://user-images.githubusercontent.com/28261578/27485118-7f5999ee-57f9-11e7-98a0-1df6443778a4.png)
Right now, all the max/min values in SWHS are defined with sub(sub()) functions which runs fine for HTML but less so for TeX.  I can change this in the example.

**Comment by szymczdm** [comment]
@AKM11 Okay, definitely follow the case study style if you can. If something else comes up, let me know. Also, you'll get reference warnings if you don't run pdflatex twice.

**Comment by niazim3** [comment]
@szymczdm Apologies, I no longer get the mentioned error. (I needed to close the editor and `Quick Build` again)

**Comment by szymczdm** [comment]
@niazim3 Good to know!

**Comment by niazim3** [comment]
This may be what @njericha was referring to earlier, but the only main discrepancy between the original CaseStudies and the SRS pdfs is the lacking `%` symbols in the **Typical Uncertainty** column of **Data Constraints**.
![image](https://user-images.githubusercontent.com/28247301/27486261-0705b0ea-57fe-11e7-80c4-03c7fc57915d.png)
![image](https://user-images.githubusercontent.com/28247301/27486269-0c563e48-57fe-11e7-9507-480580a2e0d8.png)
Possible solutions I can suggest is either Drasil generating a `\%` whenever a `%` is coded in Haskell or adding the `%` next to the words "Typical Uncertainty" in the column header like so: **Typical Uncertainty (%)**.

**Comment by JacquesCarette** [comment]
The first solution is the preferred one.  It seems like we should have a 'percentage' `Expr` (that holds a float), so that we can render it properly.  Or maybe even not stuff this in `Expr` at all, and have a new kind of `Quantity` (`UncertainQuantity` ?) which has an uncertainty field?

**Comment by szymczdm** [comment]
@JacquesCarette I think having quantities with an uncertainty field makes the most sense (since again, we'll want to be generating these tables anyway).

For the % sign, assuming it's used in more places than just the table(s), I think it should be added to our special character list (in Unicode), as it should be rendered in different ways depending on the target output.

**Comment by JacquesCarette** [comment]
@szymczdm Agreed.  So 2 issues to spin off from this.  And a new rung to the Drasil class diagram!

**Comment by danscime** [comment]
SWHS all good
